makerdao/governance-portal-v2
# Vote delegation feature branch

### What does this PR do?

Adds UI for vote delegation

### Steps for testing:

Go to https://delegations-vote.vercel.app

Create a delegate contract:
1. Go to `/delegates/me?network=kovan`
2. Click create a delegate contract (note: you can only do this once per account)
3.  Click "Delegates" in the header, you should see your contract listed on the delegates page

Delegate/undelegate MKR to a contract:
1. Go to `/delegates?network=kovan`
2. Connect wallet if not connected
3. Click "Delegate", input the amount of MKR you wish to delegate
4. Confirm
5. To undelegate, click "Undelegate" and follow the same process

Vote on executive as a delegate:
1. Use an account that has a delegate contract (see steps for creating a delegate above)
2. Make sure you have MKR delegated to your contract (you can delegate to your own contract if necessary)
3. Go to executive page, you should see the amount of MKR delegated to you at the top of the page
4. Attempt to vote on an executive
5. It should show you voted after the tx is processed

Vote on polls as a delegate:
1. Use an account that has a delegate contract (see steps for creating a delegate above)
2. Make sure you have MKR delegated to your contract (you can delegate to your own contract if necessary)
3. Go to polling page, you should see the amount of MKR delegated to you as your voting weight in the ballot box on the right as well as in the account modal if you click your address in the top right of the header
4. Attempt to vote on polls
5. UI should indicate you have voted after tx processes

### Random GIF:

![](https://media.giphy.com/media/l0MYPS5wI6kdRqsYE/giphy.gif)
